src/python/WMCore/ReqMgr/DataStructs/Request.py

```diff
@@ -188,8 +188,12 @@ def andFilterCheck(self, filterDict):
             if not isinstance(value, list):
                 value = [value]
 
-            if key in self.data:
-                if self.data[key] not in value:
+            reqValue = self.get(key)
```

Comment:
but looking here
https://github.com/ticoann/WMCore/blob/76a2f26ec11e60cf06afe4a9aa275a7eb70dd26b/src/python/WMCore/ReqMgr/DataStructs/Request.py#L122
don't we assign the whole request dict to self.data?